src/server.rs

```diff
@@ -1,38 +1,36 @@
 use super::minecraft_protocol::read_var_int;
+use super::messenger::{MessengerOperations, NewConnectionMessage};
 
-use super::packet::read;
-use super::packet_router;
 use std::env;
 use std::io;
 use std::net::TcpListener;
 use std::net::TcpStream;
 use std::thread;
+use std::io::{Cursor, Read};
 
-use super::game_state::player::PlayerStateOperations;
-use super::messenger::{MessengerOperations, NewConnectionMessage};
+use super::packet_processor::{PacketProcessorOperations, InboundPacketMessage};
 use std::sync::mpsc::Sender;
 use uuid::Uuid;
 
-pub fn listen(messenger: Sender<MessengerOperations>, player_state: Sender<PlayerStateOperations>) {
+pub fn listen(inbound_packet_processor: Sender<PacketProcessorOperations>, messenger: Sender<MessengerOperations>) {
```

Comment:
Good news here is we should basically never have to update this file again, since it now truly only cares about the things it should care about:
- reading packets from the tcp stream
- accepting new connections, assigning them connection ids, notifying the messenger about the connection